### ISSUE TITLE
显存泄漏问题

### ISSUE THREAD

**Issue by Kris-Lcq** [VDIGPKU/T-SEA#4]
作者您好，我在跑您train_optim.py这个代码的时候，随着epoch的增长，显存占用也在缓慢增长，请问作者在自己的设备上跑的时候遇到过这种问题吗？

**Comment by celinechanny** [comment]
您好，我们在训练的时候没有观察到这个现象。

有出现epoch增长之后报显存溢出导致无法继续训练的问题吗？我们之前的实验在5k个epoch时也是可以正常训练的。

如果实在不行，可以尝试在训练代码里显式地清缓存`torch.cuda.empty_cache()`。

**Comment by zhongguadepi** [comment]
> 作者您好，我在跑您train_optim.py这个代码的时候，随着epoch的增长，显存占用也在缓慢增长，请问作者在自己的设备上跑的时候遇到过这种问题吗？

你好请问你的这个问题解决了吗？我也遇到同样的问题：显存占用缓慢增长

**Comment by lesmiserable** [comment]
你好 同问这个问题 我试着在每个epoch添加torch.cuda.empty_cache() 但显存依旧会溢出

**Comment by ziyannchen** [comment]
方便report一下显卡型号这些基本信息吗？之前在Titan的卡上没有发现这个问题，我们最近打算在别的服务器复现一下这个问题。

**Comment by xiwen1** [comment]
3090ti   CUDA Version: 12.0

**Comment by lesmiserable** [comment]
您好 是在ubuntu18上 3080ti CUDA11.3

**Comment by xiwen1** [comment]
在了v100的设备上就不会出现内存泄露的问题了

**Comment by lesmiserable** [comment]
在我们的设备上还有一个现象 在V4和V4-TINY上不存在泄露问题 但在V3和V5中仍存在
附：在换用了3090和CUDA12.1后 仍是相同问题

**Comment by zhongguadepi** [comment]
> 在我们的设备上还有一个现象 在V4和V4-TINY上不存在泄露问题 但在V3和V5中仍存在 附：在换用了3090和CUDA12.1后 仍是相同问题

请问您最后解决显存泄露的这个问题了吗

**Comment by zhongguadepi** [comment]
> 在了v100的设备上就不会出现内存泄露的问题了
您好，请问最后解决这个显存泄露的问题了吗

**Comment by EinesTages** [comment]
假如有显存泄露问题可以考虑在nms的时候传入detections_with_grad.detach()

**Comment by EinesTages** [comment]
> 在了v100的设备上就不会出现内存泄露的问题了
是nms的问题，看下我的回复

**Comment by zhongguadepi** [comment]
请问您有显存泄露问题吗？可否加V 具体探讨



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年2月1日(星期四) 晚上7:38
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [VDIGPKU/T-SEA] 显存泄漏问题 (Issue #4)





 
假如有显存泄露问题可以考虑在nms的时候传入detections_with_grad.detach()

**Comment by xiwen1** [comment]
> 假如有显存泄露问题可以考虑在nms的时候传入detections_with_grad.detach()

确实有效果，十分感谢！

**Comment by zhongguadepi** [comment]
> > 如果有明显存在泄露问题可以考虑在nms的时候确定detections_with_grad.detach()
> 
> 使用YOLOv5的话这里添加确实有效果，十分感谢！
> 
> https://github.com/VDIGPKU/T-SEA/blob/7736ce8f4b66ab502a8623bab51a7b392af283da/detlib/HHDet/yolov5/api.py#L38
> 
> 
> 请问是什么原因导致了这个问题？
请问detections_with_grad.detach()代码具体添加的位置是在哪？可否加q具体探讨

**Comment by xiwen1** [comment]
> > > 如果有明显存在泄露问题可以考虑在nms的时候确定detections_with_grad.detach()
> > 
> > 
> > 使用YOLOv5的话这里添加确实有效果，十分感谢！
> > https://github.com/VDIGPKU/T-SEA/blob/7736ce8f4b66ab502a8623bab51a7b392af283da/detlib/HHDet/yolov5/api.py#L38
> > 
> > 请问是什么原因导致了这个问题？
> > 请问detections_with_grad.detach()代码具体添加的位置是在哪？可否加q具体探讨

你可以在我的profile里找到我的联系方式

**Comment by ziyannchen** [comment]
> > 假如有显存泄露问题可以考虑在nms的时候传入detections_with_grad.detach()
> 
> 使用YOLOv5的话在这里添加确实有效果，十分感谢！
> 
> https://github.com/VDIGPKU/T-SEA/blob/7736ce8f4b66ab502a8623bab51a7b392af283da/detlib/HHDet/yolov5/api.py#L38
> 
> 

有显存泄漏问题的朋友可以尝试用这个方法解决。

显存泄漏问题只观察到存在特定的机器上，目前还不明确具体的理由，欢迎有任何idea的朋友之后再评论讨论可能的理由。
